dom/src/mockDOMSource.ts

```diff
@@ -1,81 +1,81 @@
 import {StreamAdapter, DevToolEnabledSource} from '@cycle/base';
+import {DOMSourceable, GenericStream, EventsFnOptions} from './DOMSourceable';
 import {VNode} from './interfaces';
-import xsSA from '@cycle/xstream-adapter';
-import {DOMSource, EventsFnOptions} from './DOMSource';
+import xsAdapter from '@cycle/xstream-adapter';
 import xs from 'xstream';
 
-export type GenericStream = any;
-export type ElementStream = any;
-export type EventStream = any;
-
 export type MockConfig = {
   [name: string]: GenericStream | MockConfig;
   elements?: GenericStream;
 }
 
-const SCOPE_PREFIX = '___';
+const CYCLE_SOURCE = `MockedDOM`;
+const CSS_NOTATION = `.`;
+const SCOPE_PREFIX = `___`;
 
-export class MockedDOMSource implements DOMSource {
+export class MockedDOMSource implements DOMSourceable {
   private _elements: any;
 
   constructor(private _streamAdapter: StreamAdapter,
               private _mockConfig: MockConfig) {
-    if (_mockConfig.elements) {
-      this._elements = _mockConfig.elements;
-    } else {
-      this._elements = _streamAdapter.adapt(xs.empty(), xsSA.streamSubscribe);
-    }
+    this._elements = _mockConfig.elements ||
+                     _streamAdapter.adapt(xs.empty(), xsAdapter.streamSubscribe);
   }
 
   public elements(): any {
     const out: DevToolEnabledSource = this._elements;
-    out._isCycleSource = 'MockedDOM';
+    out._isCycleSource = CYCLE_SOURCE;
+
     return out;
   }
 
   public events(eventType: string, options: EventsFnOptions): any {
     const mockConfig = this._mockConfig;
     const keys = Object.keys(mockConfig);
-    const keysLen = keys.length;
-    for (let i = 0; i < keysLen; i++) {
-      const key = keys[i];
+    const keysCount = keys.length;
+    for (let idx = 0; idx < keysCount; idx++) {
+      const key = keys[idx];
       if (key === eventType) {
         const out: DevToolEnabledSource = mockConfig[key];
-        out._isCycleSource = 'MockedDOM';
+        out._isCycleSource = CYCLE_SOURCE;
+
         return out;
       }
     }
     const out: DevToolEnabledSource = this._streamAdapter.adapt(
       xs.empty(),
-      xsSA.streamSubscribe
+      xsAdapter.streamSubscribe
     );
-    out._isCycleSource = 'MockedDOM';
+    out._isCycleSource = CYCLE_SOURCE;
+
     return out;
   }
 
-  public select(selector: string): DOMSource {
+  public select(selector: string): DOMSourceable {
     const mockConfig = this._mockConfig;
     const keys = Object.keys(mockConfig);
-    const keysLen = keys.length;
-    for (let i = 0; i < keysLen; i++) {
-      const key = keys[i];
+    const keysCount = keys.length;
+    for (let idx = 0; idx < keysCount; idx++) {
```

Comment:
`i` most likely isn’t problematic in tight loops. The notion of `i`, `j`, `k`, etc. comes from maths where they are commonly used as iterator variables. FORTRAN first letter to default to integer type was `i`. Working with three loops gave the following `j`, `k`.

True, @Widdershin , `index` would be totally fine, and probably better because idx isn’t a well-known acronym for index.

I don’t really have a problem with `i` despite the fact that `i` isn’t an acronym for index, either. And what is the next acronym for index in a nested loop to be called? `j`?

Again, this is a tight loop where it’s easy to see how and why `i` is used. I simply corrected out of personal convention, because I *know* how fast code rots.